### PR TITLE
Include head requests

### PR DESCRIPTION
**Pull request by ductoanthanh**
## Note
- This will create a **breaking change** in `aws-log-replay` consumer. Previously with `lb` type, `GeneratePath` stream will only push the uri path as a string to next streams. With this upgrade, the `method` will be added with `path` in the object format. So next streams in the pipe need to read the line as an object, not a plain string.
- Format of the output of `GeneratePath` stream
```js
{
  "path": "/a.json?option=1", 
  "method": "GET", // can be GET or HEAD
  "referer": "Referer" // for 'cloudfront' type only
}
```

## Context:
- Currently all requests are replayed as GET, even though the original request might be a HEAD request. We need to include the request method in stream when generating a path via `GeneratePath`, so `RequestStream` can replay the original request with the correct request method

## Summary of changes
- Include the request method when generatePath. Change to use object format instead of array/string
- Only GET & HEAD requests are allowed to be replayed

### CODE REVIEW

**Comment by ductoanthanh** [comment]
Thinking about converting generatePath from array to string with specific keys like `path` or `method` or `referrer` instead of using array index number. It will create bigger changes but it'll definitely improve code readability. Thoughts on this?

**Comment by ductoanthanh** [comment]
Tested on flightchecks-f. HEAD requests are coming in too. 
```
SELECT *
FROM LOGS.elb_logs_30_days
WHERE dt >= '2022-08-10'
and hr >= '12'
AND stack like 'api-gl-flightchecks-f'
and method = 'HEAD'
```
Screenshot 2022-08-10 at 18 06 42

File: index.js [hunk 1]
```diff
@@ -44,21 +46,32 @@ function GeneratePath(type, keepReferer = false) {
...
           var referer = parts[9];
         }
-        // get Referer
-        if (path && referer) generatePath.push([path, referer]);
-        else generatePath.push(path);
+
+        const method = parts[5];
+        // get Referer & method
+        if (method && allowedMethods.some((m) => method.includes(m))) {
+          const obj = { path, method };
```
Comment (axrj) [inline index.js:58]: Isn't the regex match required here so the method value doesn't have an extra `"`?

**Review by axrj — APPROVED** [review]
lgtm. left one question.

File: index.js (hunk 1 above)
Comment (ductoanthanh) [reply index.js:58]: Thanks for reviewing! For Cloudfront logs, there is no extra `"` before the method so no regex is needed here.

**Review by axrj — APPROVED** [review]
great. thanks for flagging it as major version upgrade.

**Pull request merged**